_posts/2019-07-18-edge-to-kubernetes.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: post
+current: post
+cover: hacker-in-the-dark.jpg
+title: Connecting Edge IoT devices with Kubernetes
+date: 2019-07-18 08:00:00
+category: tutorials
+author: Viktor Petersson
+tags: [Open Source]
+class: post-template
+---
+
+# Bridging Edge IoT devices with Kubernetes
+
+**Required Skill Level**: Medium to Expert
+
+In my [last blog post]({{site.url}}/blog/tutorials/2019/07/15/mtls-with-nginx), we covered how to configure Nginx to use Mutual TLS (mTLS). This illustrated how mTLS can both simplify the back-end, while also make it more secure. In this article, we will take this one step further and integrate this with [Kubernetes](https://kubernetes.io/). If you haven't read the last blog post, that would be a good place to start.
+
+## Pre-requisites
+
+Before we dive into the actual configuration, we will assume that you have the following already configured:
+
+ * A Kubernetes cluster (d'oh!)
+ * A valid SSL certificate for your domain
+   * You can use [cert-manager](https://github.com/jetstack/cert-manager/) from our friends at [Jetstack](https://www.jetstack.io/) if you want to use Let's Encrypt
+ * An [Nginx Ingress Controller](https://github.com/kubernetes/ingress-nginx) configured with the SSL certificate(s)
+   * It is recommended that you use the [Nginx Helm Chart](https://github.com/helm/charts/tree/master/stable/nginx-ingress) for this
+ * Familiarity with [Helm](https://helm.sh/)
+
+## Getting started
+
+With the cluster and ingress controller up and running, you should be able to route traffic to your services. If you're not familiar with this part, we recommend that you take a look at this Katacoda's [Kubernetes Ingress training scenario](https://www.katacoda.com/courses/kubernetes/create-kubernetes-ingress-routes).
+
+Before proceeding further, we will assume that you now have an Nginx Ingress controller deployed, with some kind of service behind it that you can route traffic to in the 'foobar' namespace. In the example below, we will call this service 'foobar,' and it listens on port 80.
+
+## Configuring mTLS for the Ingress Controller
+
+Because the Nginx Ingress controller is already configured to support mTLS and subsequently does all the heavy-lifting, all we really need to do is to tell our Ingress object to use the built-in functionalities.
```

Comment:
nitpick - I prefer inbuilt over built-in unless you have a particular reason for this

Reply:
I think this is probably US vs UK english. Just like upstarts vs startups ;)